datasets/big_patent/big_patent.py

```diff
@@ -0,0 +1,156 @@
+    "f": "Mechanical Engineering; Lightning; Heating; Weapons; Blasting",
+    "g": "Physics",
+    "h": "Electricity",
+    "y": "General tagging of new or cross-sectional technology",
+}
+
+# Available versions:
+# 1.0.0 lower cased tokenized words.
+# 2.0.0 cased raw strings.
+# 2.1.0 cased raw strings (fixed).
+# TODO Add raw string versions
+
+_VERSION = "1.0.0"
+
+
+class BigPatentConfig(datasets.BuilderConfig):
+    """BuilderConfig for BigPatent."""
+
+    def __init__(self, *args, cpc_codes=None, **kwargs):
+        """BuilderConfig for BigPatent.
+        Args:
+            cpc_codes: str, cpc_codes
+            **kwargs: keyword arguments forwarded to super.
+        """
+        super().__init__(*args, version=_VERSION, **kwargs)
+        self.cpc_codes = cpc_codes
+
+
+class BigPatent(datasets.GeneratorBasedBuilder):
+    """BigPatent datasets."""
+
+    BUILDER_CONFIGS = [
+        BigPatentConfig(
+            cpc_codes=list(_CPC_DESCRIPTION),
+            name="all",
+            description="Patents under all categories.",
+        ),
+    ] + [
+        BigPatentConfig(  # pylint:disable=g-complex-comprehension
+            cpc_codes=[k],
+            name=k,
+            description=("Patents under Cooperative Patent Classification (CPC)" "{0}: {1}".format(k, v)),
+        )
+        for k, v in sorted(_CPC_DESCRIPTION.items())
+    ]
+    DEFAULT_CONFIG_NAME = "all"
+    VERSION = _VERSION
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features({_DOCUMENT: datasets.Value("string"), _SUMMARY: datasets.Value("string")}),
+            supervised_keys=(_DOCUMENT, _SUMMARY),
+            homepage="https://evasharma.github.io/bigpatent/",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        dl_path = dl_manager.download_and_extract(_URL)
+        split_types = ["train", "val", "test"]
+        extract_paths = dl_manager.extract(
+            {k: os.path.join(dl_path, "bigPatentData", k + ".tar.gz") for k in split_types}
+        )
+        extract_paths = {k: os.path.join(extract_paths[k], k) for k in split_types}
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"path": extract_paths["train"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"path": extract_paths["val"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"path": extract_paths["test"]},
+            ),
+        ]
+
+    def _generate_examples(self, path=None):
+        """Yields examples."""
+        for cpc_code in self.config.cpc_codes:
+            filenames = glob.glob(os.path.join(path, cpc_code, "*"))
+            for filename in filenames:
```

Comment:
```suggestion
            for filename in sorted(filenames):
```
we need to sort here to make the order deterministic across environments